PhotoPlusFrontend/PhotoPlus/src/app/components/category/category.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, OnInit } from '@angular/core';
+import { CategoryService } from "../../services/category/category.service";
+import { Category } from '../../models/category/category';
+import { ProductService } from "../../services/product/product.service";
+import { CartService } from 'src/app/services/cart/cart.service';
+import { Product } from '../../models/product/product';
+import { SuccessModalComponent } from '../success-modal/success-modal.component';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+
+@Component({
+  selector: 'app-category',
+  templateUrl: './category.component.html',
+  styleUrls: ['./category.component.scss']
+})
+export class CategoryComponent implements OnInit {
+
+  categories: Category[];
+  products: Product[];
+  productsToShow: Product[];
+
+  constructor( private categoryService: CategoryService, private productService: ProductService, private cartService: CartService, private modalService: NgbModal ) { }
+
+  ngOnInit(): void {
+      this.categoryService.getAll().subscribe((data: Category[]) => {
+        this.categories = data;
+      });
+          this.productService.getAll().subscribe((data: Product[]) => {
+            this.products = data;
+            this.products.forEach(element => { this.productService.getDataFromLinks(element) });
+          });
+  }
+
+  filterProducts(categoryName: String) {
+      this.productsToShow = [];
+      this.products.forEach(product => {
+        console.log(categoryName, product);
+        if (categoryName == product.category) {
+          this.productsToShow.push(product);
+        }
+      });
+    }
+
+  addToCart(product: Product) {
```

Comment:
I think you should move addToCart function to productService. We should avoid duplicate code.

Reply:
It is also used in home component.

Reply:
N O